src/microbe_stage/Microbe.Interior.cs

```diff
@@ -327,6 +337,45 @@ public void QueueEmitToxin(Compound toxinCompound)
         queuedToxinToEmit = toxinCompound;
     }
 
+    /// <summary>
+    ///   Tries to secrete slime if possible
+    /// </summary>
+    public void SecreteSlime(float delta)
+    {
+        if (SlimeJets.Count < 1)
+            return;
+
+        if (slimeSecretionCooldown > 0)
+            return;
+
+        var totalSlimeSecreted = 0.0f;
+
+        // Reset the jet factor before calculating to prevent accidental build up of speed
+        slimeJetFactor = Vector3.Zero;
+
+        // Secrete slime and gather overall force from directions determined by equipped jets
+        foreach (var jet in SlimeJets)
+        {
+            var slimeSecreted = EjectCompound(mucilage, Constants.COMPOUNDS_TO_VENT_PER_SECOND * delta,
+                -jet.GetDirection(), 2);
+            slimeJetFactor += slimeSecreted * jet.GetDirection();
+            totalSlimeSecreted += slimeSecreted;
+        }
+
+        // Once we hit zero, start a cooldown timer
+        if (totalSlimeSecreted < Constants.MUCILAGE_MIN_TO_VENT)
+            slimeSecretionCooldown = Constants.MUCILAGE_COOLDOWN_TIMER;
```

Comment:
I'm unsure about what this means: `if (totalSlimeSecreted < Constants.MUCILAGE_MIN_TO_VENT)`, shouldn't venting just not happen? It seems to me that it occurs nonetheless in the lines above with `Eject compound`.

EDIT : Oh, ok, since this runs on multiple frames I guess it is in order to just know when to stop. I kept the comment above to show my confusion, and highlight that it might need some clarification (by variable names/further comments).

Also, just note that means that if you press that with less mucilage than the min, you'll vent on one frame nonetheless I believe. Not *very impactful*, granted.

Reply:
I've revamped the approach here in my recent changes, so I don't think this problem applies anymore.